compiler/record-hessian/include/record-hessian/HessianComputer.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RECORD_HESSIAN_HESSIANCOMPUTER_H__
+#define __RECORD_HESSIAN_HESSIANCOMPUTER_H__
+
+#include "record-hessian/HessianVector.h"
+
+#include <luci/IR/CircleNode.h>
+#include <luci_interpreter/Interpreter.h>
+
+namespace record_hessian
+{
+
+class HessianComputer
```

Comment:
Does this computer calculates exact hessian value? Or does it find approximate value?

It would be better to explain how hessian value is computed in the comment (adding reference would be great).